test/Identity.FunctionalTests/Pages/Account/Register.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using AngleSharp.Dom.Html;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Identity.FunctionalTests.Account
+{
+    public class Register : HtmlPage
+    {
+        private IHtmlFormElement _registerForm;
+
+        public Register(HttpClient client, IHtmlDocument register, HtmlPageContext context)
+            : base(client, register, context)
+        {
+            _registerForm = HtmlAssert.HasForm(register);
+        }
+
+        public async Task<Index> CreateValidUser(string userName, string password)
```

Comment:
Why not call this `SubmitRegisterFormAsync` to keep the follow the convention established earlier e.g. `ClickEnableAuthenticatorLinkAsync`? The action is self-descriptive in this case rather than describing the side effect

Reply:
The pages model a state machine. The methods are the transitions, so I want to have different methods for different cases, like registering an invalid user (returning back the register view)